hypha/apply/flags/views.py

```diff
@@ -26,3 +27,6 @@ def post(self, request, type, submission_pk):
             flag.delete()
 
         return JsonResponse({"result": created})
+
+    def test_func(self):
+        return self.request.user.is_apply_staff or self.request.user.is_reviewer
```

Comment:
Would be nice to have this as a wagtail permission that you set per group/role.

Reply:
The current code will give reviewers access to update Flags for Staff. As the same `FlagSubmissionCreateView` view is used for flags of type user and staff. Looks like we should create a separate view and create an abstract class for common code. Updating it. We can add permission for flags to certain groups like staff, reviewers from admin. But will also have to add table HTML code to all dashboard templates and flag create button in all submission detail pages. In case one group gets permission like a partner then creating a flag and table should appear in their dashboard.

Reply:
[SubmissionUserFlaggedView](https://github.com/OpenTechFund/hypha/blob/d7181b723de077d311477b58763f7b5d4bdfa8ad/hypha/apply/funds/views.py#L341) also need updates, it is used when display more options appear.